### ISSUE TITLE
Add a TSV reporter

### ISSUE THREAD

**Issue by seanredmond** [pa11y/pa11y#256]
In addition to the comma-delimited (CSV) reporter, it would be useful to have a tab-delimited (TSV) reporter. There are sometimes plenty of commas in the output which is always going to make it hard to cleanly import that CSV into a spreadsheet. Using tabs is an easy way to improve that in a lot of cases.

I have a working version of this and can make a PR.

**Comment by TheJaredWilcurt** [comment]
This is a simplified version of what we use on [Koa11y](https://open-indy.github.io/Koa11y).
```js
function convertResultsToString (results, format) {
    if (format == 'json') {
        var outputJSON = results;
        outputJSON = JSON.stringify(outputJSON, null, 2);

        return outputJSON;
    } else if (format == 'csv') {
        var json2csv = require('json2csv');
        var fields = [];
        for (var key in results[0]) {
            fields.push(key);
        }
        var outputCSV = json2csv({
            'data': results,
            'fields': fields
        });

        return outputCSV;
    } else if (format == 'markdown') {
        var outputMD = '# ' + url + '\n\n';

        outputMD = outputMD + '## Results\n\n';
        var hr = '\n* * *\n\n';
        for (i = 0; i < results.length; i++) {
            var item = results[i];
            var code = '**Code:** ' + item.code + '  \n';
            var type = '**Type:** ' + item.type + '  \n';
            var typeCode = '**Type Code:** ' + item.typeCode + '  \n';
            var message = '**Message:** ' + item.message + '  \n';
            var selector = '**Selector:** `' + item.selector + '`  \n';
            var context = '**Context:**\n```\n' + item.context + '\n```\n';
            outputMD = outputMD + code + type + typeCode + message + selector + context;
            if (i < results.length - 1) {
                outputMD = outputMD + hr;
            }
        }

        return outputMD;
    } else if (format == 'xml') {
        var outputXML = '<?xml version="1.0" encoding="UTF-8"?>\n<pa11y>\n';

        for (i = 0; i < results.length; i++) {
            var current = results[i];
            var result =
                '  <result>\n' +
                '    <code>' + current.code + '</code>\n' +
                '    <type typecode="' + current.typeCode + '">' + current.type + '</type>\n' +
                '    <message>' + current.message + '</message>\n' +
                '    <selector><![CDATA[' + current.selector + ']]></selector>\n' +
                '    <context><![CDATA[' + current.context + ']]></context>\n' +
                '  </result>\n';
            outputXML = outputXML + result;
        }
        outputXML = outputXML + '</pa11y>\n';

       return outputXML;
    }
}
```

I've left out the HTML output as it's much more complex and tied in to our system.

**Comment by seanredmond** [comment]
Thanks @TheJaredWilcurt. It was dead easy to copy `reporter/csv.js` and modify it to use tabs instead. I think the use of `JSON.stringify()` makes for some messy escaping, but that's a problem for TSV and the existing CSV output, so that's probably better for a separate issue. 

I'll make a PR in a bit.

**Comment by rowanmanning** [comment]
Released as part of Pa11y 4.10.0
